plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftUnload.java

```diff
@@ -0,0 +1,239 @@
+        return RedshiftQueryRunner.builder()
+                // NOTE this can cause tests to time-out if larger tables like
+                //  lineitem and orders need to be re-created.
+                .setInitialTables(REQUIRED_TPCH_TABLES)
+                .build();
+    }
+
+    private RedshiftSplitManager createSplitManager()
+    {
+        DriverConnectionFactory driverConnectionFactory = DriverConnectionFactory.builder(new Driver(), JDBC_URL, new ExtraCredentialProvider(Optional.empty(), Optional.empty(), new StaticCredentialProvider(Optional.of(JDBC_USER), Optional.of(JDBC_PASSWORD))))
+                .build();
+        RedshiftClient redshiftClient = new RedshiftClient(
+                new BaseJdbcConfig().setConnectionUrl(JDBC_URL),
+                redshiftUnloadConfig,
+                driverConnectionFactory,
+                new JdbcStatisticsConfig(),
+                new DefaultQueryBuilder(NONE),
+                new DefaultIdentifierMapping(),
+                NONE);
+
+        return new RedshiftSplitManager(
+                redshiftClient,
+                new DefaultQueryBuilder(NONE),
+                NONE,
+                new JdbcSplitManager(redshiftClient),
+                redshiftUnloadConfig,
+                new S3FileSystemFactory(OpenTelemetry.noop(), new S3FileSystemConfig()
+                        .setAwsAccessKey(AWS_ACCESS_KEY)
+                        .setAwsSecretKey(AWS_SECRET_KEY)
+                        .setRegion(AWS_REGION)
+                        .setStreamingPartSize(DataSize.valueOf("5.5MB")), new S3FileSystemStats()),
+                Executors.newFixedThreadPool(1));
+    }
+
+    @Test
+    void testUnloadFlow()
+            throws Exception
+    {
+        JdbcTableHandle table =
+                new JdbcTableHandle(
+                        new SchemaTableName(TEST_SCHEMA, "nation"),
+                        new RemoteTableName(Optional.of(TEST_DATABASE), Optional.of(TEST_SCHEMA), "nation"),
+                        Optional.empty());
+
+        ConnectorSession session = createUnloadSession();
+        ConnectorSplitSource splitSource = redshiftSplitManager.getSplits(null, session, table, DynamicFilter.EMPTY, Constraint.alwaysTrue());
+        assertThat(splitSource).isInstanceOf(RedshiftUnloadSplitSource.class);
+
+        ConnectorSplitSource.ConnectorSplitBatch connectorSplitBatch = splitSource.getNextBatch(2).get();
+        assertThat(connectorSplitBatch.isNoMoreSplits()).isTrue();
+        assertThat(connectorSplitBatch.getSplits()).hasSize(2);
+        assertThat(connectorSplitBatch.getSplits()).allMatch(split -> split instanceof RedshiftUnloadSplit);
+        List<RedshiftUnloadSplit> redshiftUnloadSplits = connectorSplitBatch.getSplits().stream().map(split -> ((RedshiftUnloadSplit) split)).collect(toImmutableList());
+        verifyS3UnloadedFiles(redshiftUnloadSplits);
+        removeS3UnloadedFiles(redshiftUnloadSplits);
```

Comment:
I would rather just test that unloaded files are cleaned up at the end of a normal queryRunner execution

Reply:
We don't get to know unloaded path if the query is executed through normal queryRunner.

Reply:
We don't get to know exact unloaded path of a query when executed through normal queryRunner.

Reply:
I think `io.trino.operator.OperatorStats#info` should contain `io.trino.plugin.redshift.RedshiftUnloadSplit#getSplitInfo` which should show the path

Reply:
removed